Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/PExplicitGlobal.java

```diff
@@ -60,19 +61,18 @@ public class PExplicitGlobal {
     /**
      * Get a machine of a given type and index if exists, else return null.
      *
-     * @param type Machine type
-     * @param idx  Machine index
+     * @param pid Machine pid
      * @return Machine
      */
-    public static PMachine getGlobalMachine(Class<? extends PMachine> type, int idx) {
-        List<PMachine> machinesOfType = machineListByType.get(type);
+    public static PMachine getGlobalMachine(PMachineId pid) {
+        List<PMachine> machinesOfType = machineListByType.get(pid.getType());
         if (machinesOfType == null) {
             return null;
```

Comment:
When would this be null?

Reply:
When we are allocating the machine the very first time [here](https://github.com/p-org/P/blob/dev/pexplicit_checker/Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/scheduler/Scheduler.java#L276)